### PR TITLE
DDF-2046 Upgrade to Solr 6.0

### PR DESCRIPTION
**Pull request by pklinef**
#### What does this PR do?
Upgrades Solr to version 6.0

#### Who is reviewing it 
@brendan-hofmann 
@AzGoalie 

#### Choose 2 committers to review/merge the PR 
@jaymcnallie
@kcwire 

#### How should this be tested?
Verify that Solr related services such as catalog, cache, and persistence store still work properly.

#### Any background context you want to provide?
http://lucene.apache.org/solr/6_0_0/changes/Changes.html

#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-2046

#### Checklist:
- [ n/a ] Documentation Updated
- [x] Update / Add Unit Tests
- [ n/a ] Update / Add Integration Tests

### CODE REVIEW

File: platform/solr/platform-solr-server-standalone/src/main/webapp/WEB-INF/web.xml [hunk 1]
```diff
@@ -141,7 +141,7 @@
 
   <servlet-mapping>
     <servlet-name>LoadAdminUI</servlet-name>
-    <url-pattern>/admin.html</url-pattern>
+    <url-pattern>/old.html</url-pattern>
```
Comment (brendan-hofmann) [inline platform/solr/platform-solr-server-standalone/src/main/webapp/WEB-INF/web.xml]: Is `old.html` just what solr calls it? Seems like a strange name.

File: platform/solr/platform-solr-server-standalone/src/main/webapp/old.html [hunk 2]
```diff
@@ -79,8 +79,9 @@
                 
       </div>
 
-      <div class="other-ui-link">
-        Try <a class="ul" href="/solr/index.html">New UI</a><a target="_blank" href="http://wiki.apache.org/solr/AngularUI">&nbsp;<span class="help"></span></a>
+      <div class="old-ui-warning">
+        THIS USER INTERFACE IS DEPRECATED. Please use the current UI <a class="ul" href="/solr/">here</a>
```
Comment (brendan-hofmann) [inline platform/solr/platform-solr-server-standalone/src/main/webapp/old.html]: Guess this answers my question. :) What's the value in keeping the old ui though, vs simply redirecting it to the new ui (maybe still with a warning)?

File: catalog/core/catalog-core-solr/pom.xml [hunk 3]
```diff
@@ -128,7 +123,7 @@
                             <include>ddf.platform.solr:solr-xpath</include>
                             <include>org.apache.lucene:lucene-core</include>
                             <include>ddf.catalog.core:catalog-core-api-impl</include>
-                            <include>com.spatial4j:spatial4j</include>
+                            <include>org.locationtech.spatial4j:spatial4j</include>
```
Comment (kcwire) [inline catalog/core/catalog-core-solr/pom.xml]: Do you need to add `org.locationtech.spatial4j` as a dependency? not sure where maven shade is getting it's version and such without the explicit declaration.

File: catalog/pom.xml [hunk 4]
```diff
@@ -43,7 +43,7 @@
             Remove restlet.version property and restlet dependencies from the dependencyManagement section when
             Solr version is updated (see DDF-1203).
         -->
-        <solr.version>5.5.0</solr.version>
+        <solr.version>6.0.0</solr.version>
```
Comment (kcwire) [inline catalog/pom.xml]: Can we remove the restlet version now?  It seems it was added for Solr 4.10.

**Comment by kcwire** [comment]
👍

File: catalog/core/catalog-core-solr/pom.xml (hunk 3 above)
Comment (pklinef) [inline catalog/core/catalog-core-solr/pom.xml]: It figures it out from the Solr dependencies.  This should make it easier to do Solr updates in the future. You can see the same thing in the Solr dependencies project under platform/solr.

File: catalog/pom.xml (hunk 4 above)
Comment (pklinef) [inline catalog/pom.xml]: Looks like the restlet version and the solr saxon version are no longer used.  I removed both.

**Comment by jaymcnallie** [comment]
LGTM 👍

**Pull request merged**